x/btcstkconsumer/keeper/consumer_registry.go

```diff
@@ -61,23 +54,14 @@ func (k Keeper) IsCosmosConsumer(ctx context.Context, consumerID string) (bool,
 
 // GetAllRegisteredConsumerIDs gets all consumer IDs that registered to Babylon
 func (k Keeper) GetAllRegisteredConsumerIDs(ctx context.Context) []string {
-	consumerIDs := []string{}
-	iter := k.consumerRegistryStore(ctx).Iterator(nil, nil)
-	defer iter.Close()
-
-	for ; iter.Valid(); iter.Next() {
-		consumerIDBytes := iter.Key()
-		consumerID := string(consumerIDBytes)
+	var consumerIDs []string
+	err := k.ConsumerRegistry.Walk(ctx, nil, func(consumerID string, _ types.ConsumerRegister) (bool, error) {
 		consumerIDs = append(consumerIDs, consumerID)
+		return false, nil
+	})
+	if err != nil {
+		// Return empty list on error rather than panicking
+		return []string{}
 	}
```

Comment:
Hmm why not panick here? If `Walk` failed then there might be sth really bad in DB already